Assets/Scripts/SpellCaster.cs

```diff
@@ -0,0 +1,136 @@
+using System.Collections;
+
+public class SpellCaster : MonoBehaviour
+{
+    [Header("Spell System")]
+    public SpellInventory spellInventory;
+    public Transform firePoint;
+    public float castDelay = 0.2f;
+    
+    [Header("Current Spell")]
+    private int currentSpellIndex = 0;
+    
+    private SpellData currentSpell;
+    private Animator animator;
+    private bool isCasting = false;
+    
+    private void Awake()
+    {
+        animator = GetComponent<Animator>();
+    }
+    
+    private void Start()
+    {
+        if (spellInventory != null && spellInventory.GetAllSpells().Count > 0)
+        {
+            SelectSpell(0);
+        }
+    }
+    
+    private void SelectSpell(int index)
+    {
+        var spells = spellInventory.GetAllSpells();
+        if (index >= 0 && index < spells.Count)
+        {
+            currentSpellIndex = index;
+            currentSpell = spells[index].spellData;
+            Debug.Log($"Selected spell: {currentSpell.spellName}");
+        }
+    }
+    
+    private void SelectSpellByName(string spellName)
+    {
+        var spells = spellInventory.GetAllSpells();
+        for (int i = 0; i < spells.Count; i++)
+        {
+            if (spells[i].spellData.spellName.Equals(spellName, System.StringComparison.OrdinalIgnoreCase))
+            {
+                SelectSpell(i);
+                return;
+            }
+        }
+    }
+    
+    public void SelectAttackSpell()
+    {
+        SelectSpellByName("Attack");
+    }
+    
+    public void SelectStunSpell()
+    {
+        SelectSpellByName("Stun");
+    }
+    
+    private bool CanCastCurrentSpell()
+    {
+        return currentSpell != null && 
+               spellInventory.CanCastSpell(currentSpell) && 
+               !isCasting;
+    }
+    
+    public void CastCurrentSpell()
+    {
+        if (CanCastCurrentSpell())
+        {
+            StartCoroutine(CastSpellCoroutine());
+        }
+        else
+        {
+            Debug.Log($"Cannot cast {currentSpell?.spellName}: " +
+                     $"Cooldown remaining: {spellInventory.GetSpellCooldownRemaining(currentSpell):F1}s");
+        }
+    }
+    
+    private IEnumerator CastSpellCoroutine()
+    {
+        isCasting = true;
+        
+        // Play cast animation
+        if (animator != null)
+        {
+            animator.SetInteger(PlayerAnimationConstants.Accessor, PlayerAnimationConstants.Cast);
+        }
+        
+        // Wait for cast delay
+        yield return new WaitForSeconds(castDelay);
+        
+        // Cast the spell
+        if (spellInventory.CastSpell(currentSpell))
+        {
+            FireSpellProjectile();
+            Debug.Log($"Cast {currentSpell.spellName}!");
+        }
+        
+        isCasting = false;
+    }
+    
+    private void FireSpellProjectile()
+    {
+        if (currentSpell.spellPrefab != null && firePoint != null)
+        {
+            GameObject projectile = SpawnManager.Instantiate(
```

Comment:
Instead of Instantiate, call InstantiateAndNotify so the Looper can observe it. Also set the flip here